device/transport/amqp/src/amqp.ts

```diff
@@ -236,6 +236,11 @@ export class Amqp extends EventEmitter implements Client.Transport {
           disableMethods: (callback) => {
             // if we are disconnected the C2D link is already detached.
             callback();
+          },
+          disconnectError: (err, callback) => {
+            /*Codes_SRS_NODE_DEVICE_AMQP_16_080: [if the handler specified in the `setDisconnectHandler` call is called while the `Amqp` object is disconnected, the call shall be ignored.]*/
+            debug('ignoring disconnectError because already disconnected: ' + err.toString());
```

Comment:
>ignoring [](start = 19, length = 8)

does this swallow an error that we might want to forward up to higher layers?

Reply:
it is on purpose. if we're already disconnected, we're already in the right state and the higher layers have already been notified (using the _onEnter callback calling/event emitting code). we should not be hitting that but i want to be defensive and log it just in case amqp10 does something funky. wouldn't be the first time we have a weird race condition.

---
In reply to: [175762815](https://github.com/Azure/azure-iot-sdk-node/pull/254#discussion_r175762815) [](ancestors = 175762815)